### PR TITLE
[test] Introduce Github action for CodeChecker tests

### PR DESCRIPTION
**Pull request by csordasmarton**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
+name: codechecker-tests
+
+# Triggers the workflow on push or pull request events.
+on: [push, pull_request]
```
Comment (gyorb) [inline .github/workflows/test.yml:4]: are there triggers for merging to master master branch?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Install dependencies
+        run: |
+          pip install $(grep -iE "pylint|pycodestyle" analyzer/requirements_py/dev/requirements.txt)
+      - name: Run tests
+        run: make pylint pycodestyle
```
Comment (gyorb) [inline .github/workflows/test.yml:21]: are there any javascript linters?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+      - name: Run analyzer tests
+        run: |
+          make -C analyzer \
+            test_unit \
+            test_functional \
+            test_tu_collector \
+            test_merge_clang_extdef_mappings \
+            test_statistics_collector \
+            test_build_logger
```
Comment (gyorb) [inline .github/workflows/test.yml:49]: where are the tools (plist to html, report_converters, tu_collector ...) tests executed?
Maybe a separate step would be good for them?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      - name: Run tests
+        env:
+          PGPASSWORD: postgres
+        run: |
+          export PGPASSFILE=$HOME/.pgpass
+
+          make pip_dev_deps
+          BUILD_UI_DIST=NO make package
+
+          make -C tools/plist_to_html test
```
Comment (gyorb) [inline .github/workflows/test.yml:95]: Maybe the tools (plist to html, report_converters, tu_collector ...) tests could be executed as a separate job?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            test_unit \
+            test_functional \
+            test_tu_collector \
+            test_merge_clang_extdef_mappings \
+            test_statistics_collector \
+            test_build_logger
+
+  web:
+    name: Web
+    runs-on: ubuntu-latest
```
Comment (gyorb) [inline .github/workflows/test.yml:53]: Would it be better to select some specific ubuntu version? What is the latest used here? LTS latest (20.04)?
The install_deps.sh assumes that the distribution is ubuntu bionic.

**Review by gyorb — CHANGES_REQUESTED** [review]

**Review by csordasmarton — COMMENTED** [review]
My comment for https://github.com/Ericsson/codechecker/pull/3066#discussion_r533382675 comment: we run the JavaScript linters in the GUI related job. For this we have to install node and also run the npm install command which takes too much time. In the gui job it is necessary to install these, so this is the reason why I moved GUI linter test to this job and not to the separate linter job.

**Review by gyorb — APPROVED** [review]

**Pull request merged**